CesiumIonClient/src/fillWithRandomBytes.cpp

```diff
@@ -0,0 +1,63 @@
+#define _CRT_RAND_S
+#include "fillWithRandomBytes.h"
+
+#include <cassert>
+
+// When WINAPI_FAMILY_PARTITION && WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP)
+// is true, this is a Univeral Windows Platform build. csprng doesn't work on
+// UWP. So we use the Windows-only rand_s function instead.
+
+#ifdef WINAPI_FAMILY
+#if WINAPI_FAMILY == WINAPI_FAMILY_PC_APP ||                                   \
+    WINAPI_FAMILY == WINAPI_FAMILY_PHONE_APP
+#define IS_UWP 1
+#else
+#define IS_UWP 0
+#endif
+#else
+#define IS_UWP 0
+#endif
+
+#if IS_UWP
+#include <cstdlib>
+#else
+#include <duthomhas/csprng.hpp>
+#endif
+
+namespace CesiumIonClient {
+
+void fillWithRandomBytes(const gsl::span<uint8_t>& buffer) {
+#if IS_UWP
+  size_t i = 0;
+  if (buffer.size() >= sizeof(uint32_t)) {
+    for (; i <= buffer.size() - sizeof(std::uint32_t);
+         i += sizeof(std::uint32_t)) {
+      std::uint32_t r;
+      if (rand_s(&r) != 0) {
+        throw std::exception("Failed to generate random numbers.");
```

Comment:
This is extremely unlikely to fail, short of something catastrophically wrong at the operating system level. If it does, it's fatal, which is why I was ok throwing. In some contexts (Unreal Engine shipping builds, for example), throwing an exception will result in immediate process termination.

The CSPRNG path is the same, but it's hidden inside that library. If something goes wrong and its unable to generate secure random numbers, it will throw an exception.

Reply:
You are correct, CSPRNG does indeed throw...
![image](https://github.com/CesiumGS/cesium-native/assets/130494071/75cafb58-1383-49da-a4d7-c77a127e1a4b)

when it said it wouldn't. Hmmm
![image](https://github.com/CesiumGS/cesium-native/assets/130494071/7ad7df05-ed28-4f69-b1b1-8208d1758374)

Reply:
Agreed. Not being able to generate CS random numbers should be fatal. I'm glad that both paths throw in this case.